processor/src/it/java/com/linecorp/decaton/processor/CoreFunctionalityTest.java

```diff
@@ -153,17 +154,17 @@ public void testSingleThreadProcessing() throws Exception {
         // Note that this processing semantics is not be considered as Decaton specification which users can rely on.
         // Rather, this is just a expected behavior based on current implementation when we set concurrency to 1.
         ProcessingGuarantee noDuplicates = new ProcessingGuarantee() {
-            private final Map<String, List<TestTask>> produced = new HashMap<>();
-            private final Map<String, List<TestTask>> processed = new HashMap<>();
+            private final Map<ByteBuffer, List<TestTask>> produced = new HashMap<>();
```

Comment:
ByteBuffer is mutable and its equals/hashcode methods depend on position/limit/remaining. Meaning, reading from it will affect findability in the map.

I don't know if is this overkill, but what about defining a new wrapper type, that
simply delegates equals/hashCode to Arrays.equals / Arrays.hashCode?

It could also cache hashCode (like String does)

And have a toString that just attempts to decode it with UTF-8